### PR TITLE
[Naming] Register RenameForeachValueVariableToMatchExprVariableRector to naming config set

### PR DESCRIPTION
**Pull request by samsonasik**

### CODE REVIEW

**Comment by samsonasik** [comment]
All checks have passed 🎉 @TomasVotruba I think it is ready.

**Comment by samsonasik** [comment]
rebased.

**Comment by samsonasik** [comment]
@TomasVotruba the `rector-ci (rules)` action seems hanging : 

```bash
rector-ci (rules) Expected — Waiting for status to be reported
```

other checks are passed 🎉 I think it is ready.

**Comment by samsonasik** [comment]
All checks have passed 🎉 @TomasVotruba I think it is ready.

File: packages/better-php-doc-parser/src/PhpDocInfo/PhpDocInfo.php [hunk 1]
```diff
@@ -282,20 +282,20 @@ public function getByType(string $type): ?PhpDocNode
     {
         $this->ensureTypeIsTagValueNode($type, __METHOD__);
 
-        foreach ($this->phpDocNode->children as $phpDocChildNode) {
-            if (is_a($phpDocChildNode, $type, true)) {
-                return $phpDocChildNode;
+        foreach ($this->phpDocNode->children as $child) {
```
Comment (TomasVotruba) [inline packages/better-php-doc-parser/src/PhpDocInfo/PhpDocInfo.php:285]: This information should be kept. It's a huge tell about the type compared to just `child`

File: packages/better-php-doc-parser/src/PhpDocParser/ClassAnnotationMatcher.php [hunk 2]
```diff
@@ -76,12 +76,12 @@ private function resolveFullyQualifiedClass(array $uses, Node $node, string $tag
     private function matchFullAnnotationClassWithUses(string $tag, array $uses): ?string
     {
         foreach ($uses as $use) {
-            foreach ($use->uses as $useUse) {
-                if (! $this->isUseMatchingName($tag, $useUse)) {
+            foreach ($use->uses as $use) {
```
Comment (TomasVotruba) [inline packages/better-php-doc-parser/src/PhpDocParser/ClassAnnotationMatcher.php:79]: Here is foreached variable override, it should be skipped too.

After this foreach, the original `$use` might be overriden by iterated `$use`

File: packages/phpstan-static-type-mapper/src/TypeMapper/IterableTypeMapper.php [hunk 3]
```diff
@@ -87,17 +87,17 @@ private function convertUnionArrayTypeNodesToArrayTypeOfUnionTypeNodes(
         UnionTypeNode $unionTypeNode
     ): AttributeAwareUnionTypeNode {
         $unionedArrayType = [];
-        foreach ($unionTypeNode->types as $unionedType) {
-            if ($unionedType instanceof UnionTypeNode) {
```
Comment (TomasVotruba) [inline packages/phpstan-static-type-mapper/src/TypeMapper/IterableTypeMapper.php:91]: This special case should be kept, as `unionedType` states clearly that is iterated value. While `type` is just one main type

File: packages/better-php-doc-parser/src/PhpDocInfo/PhpDocInfo.php (hunk 1 above)
Comment (samsonasik) [reply packages/better-php-doc-parser/src/PhpDocInfo/PhpDocInfo.php]: Fixed by skip property fetch not from `ThisType`

File: packages/better-php-doc-parser/src/PhpDocParser/ClassAnnotationMatcher.php (hunk 2 above)
Comment (samsonasik) [reply packages/better-php-doc-parser/src/PhpDocParser/ClassAnnotationMatcher.php]: Fixed by skip property fetch not from `ThisType`

File: packages/phpstan-static-type-mapper/src/TypeMapper/IterableTypeMapper.php (hunk 3 above)
Comment (samsonasik) [reply packages/phpstan-static-type-mapper/src/TypeMapper/IterableTypeMapper.php]: Fixed by skip property fetch not from `ThisType`

**Comment by samsonasik** [comment]
All checks have passed 🎉 @TomasVotruba I think it is ready.

File: rules/privatization/src/Rector/Class_/ChangeReadOnlyVariableWithDefaultValueToConstantRector.php [hunk 4]
```diff
@@ -173,8 +173,10 @@ private function filterOutUniqueNames(array $assigns): array
         }
```
Comment (TomasVotruba) [inline rules/privatization/src/Rector/Class_/ChangeReadOnlyVariableWithDefaultValueToConstantRector.php:173]: This method can be extracted with return type to reach the same goal.

Comment (samsonasik) [reply rules/privatization/src/Rector/Class_/ChangeReadOnlyVariableWithDefaultValueToConstantRector.php]: implemented.

**Comment by samsonasik** [comment]
All checks have passed 🎉 @TomasVotruba I think it is ready.

**Comment by samsonasik** [comment]
@TomasVotruba the Github Actions seems pending `Expected — Waiting for status to be reported` not started.

**Comment by samsonasik** [comment]
All checks have passed 🎉 @TomasVotruba I think it is ready.

**Comment by samsonasik** [comment]
rebased.

**Comment by samsonasik** [comment]
All checks have passed 🎉 @TomasVotruba I think it is ready.

File: packages/better-php-doc-parser/tests/PhpDocParser/TagValueNodeReprint/TagValueNodeReprintTest.php [hunk 5]
```diff
@@ -41,9 +41,9 @@ public function provideData(): Iterator
     {
         foreach ($this->getDirectoriesByTagValueNodes() as $tagValueNode => $directory) {
             $filesInDirectory = $this->findFilesFromDirectory($directory);
-            foreach ($filesInDirectory as $fileInfos) {
-                foreach ($fileInfos as $fileInfo) {
-                    yield [$fileInfo, $tagValueNode];
+            foreach ($filesInDirectory as $singleFilesInDirectory) {
+                foreach ($singleFilesInDirectory as $singleSingleFilesInDirectory) {
+                    yield [$singleSingleFilesInDirectory, $tagValueNode];
```
Comment (TomasVotruba) [inline packages/better-php-doc-parser/tests/PhpDocParser/TagValueNodeReprint/TagValueNodeReprintTest.php:46]: This should be fixer. The original foreach variable seems wrong, causing broken naming

File: packages/family-tree/src/NodeAnalyzer/ClassChildAnalyzer.php [hunk 6]
```diff
@@ -24,18 +24,19 @@ public function hasChildClassMethod(ClassReflection $classReflection, string $me
     {
         $childrenClassReflections = $this->familyRelationsAnalyzer->getChildrenOfClassReflection($classReflection);
 
-        foreach ($childrenClassReflections as $childClassReflection) {
-            if (! $childClassReflection->hasMethod($methodName)) {
+        foreach ($childrenClassReflections as $childrenClassReflection) {
+            $childrenClassReflectionHasMethod = $childrenClassReflection->hasMethod($methodName);
```
Comment (TomasVotruba) [inline packages/family-tree/src/NodeAnalyzer/ClassChildAnalyzer.php:28]: This method should be inlined, and not on own line. It returns bool

File: packages/family-tree/src/NodeAnalyzer/PropertyUsageAnalyzer.php [hunk 7]
```diff
@@ -75,8 +75,8 @@ public function isPropertyFetchedInChildClass(Property $property): bool
 
         $childrenClassReflections = $this->familyRelationsAnalyzer->getChildrenOfClassReflection($classReflection);
 
-        foreach ($childrenClassReflections as $childClassReflection) {
-            $childClass = $this->nodeRepository->findClass($childClassReflection->getName());
+        foreach ($childrenClassReflections as $childrenClassReflection) {
```
Comment (TomasVotruba) [inline packages/family-tree/src/NodeAnalyzer/PropertyUsageAnalyzer.php:78]: Children to child was correct here, should be kept

File: packages/node-collector/src/NodeCollector/ParsedNodeCollector.php [hunk 8]
```diff
@@ -210,12 +210,12 @@ public function findNewsByClass(string $className): array
     {
         $newsByClass = [];
 
-        foreach ($this->news as $new) {
-            if (! $this->nodeNameResolver->isName($new->class, $className)) {
+        foreach ($this->news as $singleNews) {
+            if (! $this->nodeNameResolver->isName($singleNews->class, $className)) {
```
Comment (TomasVotruba) [inline packages/node-collector/src/NodeCollector/ParsedNodeCollector.php:214]: This should be kept, as it's not a typical news as in '"today's news"

File: packages/testing/src/PHPUnit/AbstractRectorTestCase.php [hunk 9]
```diff
@@ -203,23 +203,26 @@ protected function doTestExtraFile(string $expectedExtraFileName, string $expect
...
-            $printedFileContent = $this->betterStandardPrinter->prettyPrintFile($addedFileWithNodes->getNodes());
+            $printedFileContent = $this->betterStandardPrinter->prettyPrintFile($addedFilesWithNode->getNodes());
             $this->assertStringEqualsFile($expectedExtraContentFilePath, $printedFileContent);
             return;
         }
 
         $movedFilesWithContent = $this->removedAndAddedFilesCollector->getMovedFileWithContent();
-        foreach ($movedFilesWithContent as $movedFileWithContent) {
-            if (! Strings::endsWith($movedFileWithContent->getNewPathname(), $expectedExtraFileName)) {
+        foreach ($movedFilesWithContent as $singleMovedFilesWithContent) {
```
Comment (TomasVotruba) [inline packages/testing/src/PHPUnit/AbstractRectorTestCase.php:217]: This should be handled differently, as files goes to file.

File: src/Util/StaticRectorStrings.php [hunk 10]
```diff
@@ -93,9 +93,9 @@ public static function removePrefixes(string $value, array $prefixesToRemove): s
      */
     public static function removeSuffixes(string $value, array $suffixesToRemove): string
     {
-        foreach ($suffixesToRemove as $prefixToRemove) {
-            if (Strings::endsWith($value, $prefixToRemove)) {
-                $value = Strings::substring($value, 0, -Strings::length($prefixToRemove));
+        foreach ($suffixesToRemove as $singleSuffixesToRemove) {
```
Comment (TomasVotruba) [inline src/Util/StaticRectorStrings.php:96]: Here it should be kept too, as suffixes to suffix

File: packages/better-php-doc-parser/tests/PhpDocParser/TagValueNodeReprint/TagValueNodeReprintTest.php (hunk 5 above)
Comment (samsonasik) [reply packages/better-php-doc-parser/tests/PhpDocParser/TagValueNodeReprint/TagValueNodeReprintTest.php:46]: "filesInDirectory" when singularized with Inflector service result same value, so use "single" prefix, any other way to singularize it?

File: packages/family-tree/src/NodeAnalyzer/ClassChildAnalyzer.php (hunk 6 above)
Comment (samsonasik) [reply packages/family-tree/src/NodeAnalyzer/ClassChildAnalyzer.php:28]: this seems automatically moved by rector as not from This type

File: packages/family-tree/src/NodeAnalyzer/PropertyUsageAnalyzer.php (hunk 7 above)
Comment (samsonasik) [reply packages/family-tree/src/NodeAnalyzer/PropertyUsageAnalyzer.php:78]: any way to singularize except using Inflector service for variable?

File: packages/node-collector/src/NodeCollector/ParsedNodeCollector.php (hunk 8 above)
Comment (samsonasik) [reply packages/node-collector/src/NodeCollector/ParsedNodeCollector.php:214]: any service other than Inflector that can handle it?

File: packages/testing/src/PHPUnit/AbstractRectorTestCase.php (hunk 9 above)
Comment (samsonasik) [reply packages/testing/src/PHPUnit/AbstractRectorTestCase.php:217]: is Inflector service need extract camel case with end "s" or "es" on first part to check it?

File: src/Util/StaticRectorStrings.php (hunk 10 above)
Comment (samsonasik) [reply src/Util/StaticRectorStrings.php:96]: is Inflector service need extract camel case with end "s" or "es" on first part to check it?

File: packages/better-php-doc-parser/tests/PhpDocParser/TagValueNodeReprint/TagValueNodeReprintTest.php (hunk 5 above)
Comment (TomasVotruba) [reply packages/better-php-doc-parser/tests/PhpDocParser/TagValueNodeReprint/TagValueNodeReprintTest.php:46]: The issue is `findFilesFromDirectory()` does not return `file[]`, but `file[][]`.

The test must be refactorted to `findFilesFromDirectory()` returns `file[]`. The inflection would work.

Another solution is to skip this file for this rule and resolve this later, as it requires more refactoring unrelated to naming.

File: packages/family-tree/src/NodeAnalyzer/ClassChildAnalyzer.php (hunk 6 above)
Comment (TomasVotruba) [reply packages/family-tree/src/NodeAnalyzer/ClassChildAnalyzer.php:28]: The bool return should work for both `ThisType` and `ObjectType`.

```php
if ($this->is...()) {
}


if ($item->is...()) {
}
```

I've noticed similar behavior on any variable rename. It seems the variable looses type and returns `MixedType`. That's why it's changed by another Rector rule like this.

It should not be changed.

File: packages/family-tree/src/NodeAnalyzer/PropertyUsageAnalyzer.php (hunk 7 above)
Comment (TomasVotruba) [reply packages/family-tree/src/NodeAnalyzer/PropertyUsageAnalyzer.php:78]: Nothing I'm aware of. Maybe current Inflector will need modification to cover children/child prefix

File: packages/testing/src/PHPUnit/AbstractRectorTestCase.php (hunk 9 above)
Comment (TomasVotruba) [reply packages/testing/src/PHPUnit/AbstractRectorTestCase.php:217]: Hmm... maybe. Or we could rename `MovedFileWithContent` to comply with this naming standard.
Any ideas?

File: src/Util/StaticRectorStrings.php (hunk 10 above)
Comment (TomasVotruba) [reply src/Util/StaticRectorStrings.php:96]: Yes

File: packages/better-php-doc-parser/tests/PhpDocParser/TagValueNodeReprint/TagValueNodeReprintTest.php (hunk 5 above)
Comment (samsonasik) [reply packages/better-php-doc-parser/tests/PhpDocParser/TagValueNodeReprint/TagValueNodeReprintTest.php:46]: I will try refactor it.

File: packages/family-tree/src/NodeAnalyzer/ClassChildAnalyzer.php (hunk 6 above)
Comment (samsonasik) [reply packages/family-tree/src/NodeAnalyzer/ClassChildAnalyzer.php:28]: I will check.

File: packages/family-tree/src/NodeAnalyzer/PropertyUsageAnalyzer.php (hunk 7 above)
Comment (samsonasik) [reply packages/family-tree/src/NodeAnalyzer/PropertyUsageAnalyzer.php:78]: I will try add it.

File: packages/testing/src/PHPUnit/AbstractRectorTestCase.php (hunk 9 above)
Comment (samsonasik) [reply packages/testing/src/PHPUnit/AbstractRectorTestCase.php:217]: It seems we need to extends the `Doctrine\Inflector\Inflector` service as I realized it came from vendor.

File: src/Util/StaticRectorStrings.php (hunk 10 above)
Comment (samsonasik) [reply src/Util/StaticRectorStrings.php:96]: It seems we need to extends the `Doctrine\Inflector\Inflector` service as I realized it came from vendor.

Comment (TomasVotruba) [reply src/Util/StaticRectorStrings.php:96]: Great :+1: go for it

File: packages/testing/src/PHPUnit/AbstractRectorTestCase.php (hunk 9 above)
Comment (samsonasik) [reply packages/testing/src/PHPUnit/AbstractRectorTestCase.php]: implemented.

File: src/Util/StaticRectorStrings.php (hunk 10 above)
Comment (samsonasik) [reply src/Util/StaticRectorStrings.php]: implemented.

File: packages/node-collector/src/NodeCollector/ParsedNodeCollector.php (hunk 8 above)
Comment (samsonasik) [reply packages/node-collector/src/NodeCollector/ParsedNodeCollector.php]: implemented.

File: packages/family-tree/src/NodeAnalyzer/PropertyUsageAnalyzer.php (hunk 7 above)
Comment (samsonasik) [reply packages/family-tree/src/NodeAnalyzer/PropertyUsageAnalyzer.php]: implemented.

File: packages/family-tree/src/NodeAnalyzer/ClassChildAnalyzer.php (hunk 6 above)
Comment (samsonasik) [reply packages/family-tree/src/NodeAnalyzer/ClassChildAnalyzer.php]: updated.

File: packages/better-php-doc-parser/tests/PhpDocParser/TagValueNodeReprint/TagValueNodeReprintTest.php (hunk 5 above)
Comment (samsonasik) [reply packages/better-php-doc-parser/tests/PhpDocParser/TagValueNodeReprint/TagValueNodeReprintTest.php]: I added InflectorSingularResolver service to handle it https://github.com/rectorphp/rector/pull/5696/commits/a793125018f0f137636659b5d24279e4ae5c276a

**Comment by samsonasik** [comment]
All checks have passed 🎉 @TomasVotruba I think it is ready.

File: rules/dead-code/src/Rector/FunctionLike/RemoveDuplicatedIfReturnRector.php [hunk 11]
```diff
@@ -104,12 +104,12 @@ public function refactor(Node $node): ?Node
             return null;
         }
 
-        foreach ($ifWithOnlyReturnsByHash as $stmts) {
+        foreach ($ifWithOnlyReturnsByHash as $ifWithOnlyReturnByHash) {
```
Comment (TomasVotruba) [inline rules/dead-code/src/Rector/FunctionLike/RemoveDuplicatedIfReturnRector.php:107]: This is a special naming "x by y", e.g. methosByClass.
The result of this single name should be left part infront of by, e.g. methods

Comment (samsonasik) [reply rules/dead-code/src/Rector/FunctionLike/RemoveDuplicatedIfReturnRector.php:107]: implemented https://github.com/rectorphp/rector/pull/5696/commits/9a4aa796e13d7bd4dc822440dc784b8d334c04db

Comment (samsonasik) [reply rules/dead-code/src/Rector/FunctionLike/RemoveDuplicatedIfReturnRector.php]: regenerated https://github.com/rectorphp/rector/pull/5696/commits/594db778483b9e4e090a37333e0efbf737654fcd

**Comment by samsonasik** [comment]
All checks have passed 🎉 @TomasVotruba I think it is ready.

**Comment by samsonasik** [comment]
rebased.

**Comment by samsonasik** [comment]
All checks have passed 🎉 @TomasVotruba I think it is ready.

File: packages/node-type-resolver/src/Reflection/BetterReflection/SourceLocatorProvider/DynamicSourceLocatorProvider.php [hunk 12]
```diff
@@ -54,8 +54,8 @@ public function provide(): SourceLocator
             $sourceLocators[] = new OptimizedSingleFileSourceLocator($this->fileNodesFetcher, $file);
         }
 
-        foreach ($this->filesByDirectory as $files) {
-            $sourceLocators[] = new OptimizedDirectorySourceLocator($this->fileNodesFetcher, $files);
+        foreach ($this->filesByDirectory as $file) {
```
Comment (TomasVotruba) [inline packages/node-type-resolver/src/Reflection/BetterReflection/SourceLocatorProvider/DynamicSourceLocatorProvider.php:57]: This case `x by y` needs to be covered.

```php
foreach ($files By Directory as $files) {
    // ...
}
```

Comment (samsonasik) [reply packages/node-type-resolver/src/Reflection/BetterReflection/SourceLocatorProvider/DynamicSourceLocatorProvider.php:57]: implemented https://github.com/rectorphp/rector/pull/5696/commits/f589ba3949addb09901e08a7a0e0ed5841c7e52d

File: rules/dead-code/src/Rector/FunctionLike/RemoveOverriddenValuesRector.php [hunk 13]
```diff
@@ -175,27 +175,28 @@ private function resolveNodesToRemove(array $assignedVariableNames, array $nodes
         foreach ($assignedVariableNames as $assignedVariableName) {
             $previousNode = null;
 
-            foreach ($nodesByTypeAndPosition as $nodeByTypeAndPosition) {
-                $variableNode = $nodeByTypeAndPosition->getVariableNode();
+            foreach ($nodesByTypeAndPosition as $nodes) {
+                $variableNode = $nodes->getVariableNode();
```
Comment (samsonasik) [inline rules/dead-code/src/Rector/FunctionLike/RemoveOverriddenValuesRector.php:179]: @TomasVotruba I think we need to check the static type of the variable before use of previous `By` usage, as `$nodesByTypeAndPosition`, the `$nodes` is a single Values object, not array.

Comment (samsonasik) [reply rules/dead-code/src/Rector/FunctionLike/RemoveOverriddenValuesRector.php:179]: @TomasVotruba I tried to check the value var of foreach is Array Type, but it always got Mixed Type at https://github.com/rectorphp/rector/pull/5696/commits/adb9f767f20ea2544e5ccfc9cfe361ecc929912a , do you have idea how to get it? Thank you.

Comment (samsonasik) [reply rules/dead-code/src/Rector/FunctionLike/RemoveOverriddenValuesRector.php:179]: I reverted the type of value var of the foreach, so use as `nodes` as is.

**Comment by samsonasik** [comment]
All checks have passed 🎉 @TomasVotruba I think it is ready.

**Comment by TomasVotruba** [comment]
Thank you for all the fixes.

Let's give it a try and ship it 👍

**Pull request merged**